### PR TITLE
Maybe cache fix

### PR DESCRIPTION
**Pull request by connorsmith256**

### CODE REVIEW

File: steps/rust-caching.yml [hunk 1]
```diff
@@ -25,17 +25,16 @@ steps:
           Write-Host "##vso[task.setvariable variable=homedir]$env:HOME"
         displayName: Get Home directory (windows)
         condition: eq(variables['Agent.OS'], 'Windows_NT')
-  - ${{ if parameters.cacheRegistry }}:
```
Comment (connorsmith256) [inline steps/rust-caching.yml:28]: Not sure if this was needed/here for cleanliness or what

File: steps/rust-caching.yml [hunk 2]
```diff
@@ -25,17 +25,16 @@ steps:
...
       - task: Cache@2
         inputs:
-          key: '"rust-${{ parameters.projectName }}-cache" | "$(Agent.OS)" | "${{ parameters.workingDirectory }}/Cargo.toml"'
+          key: '"rust-${{ parameters.projectName }}-cache" | "$(Agent.OS)" | ${{ parameters.workingDirectory }}/Cargo.toml'
           path: $(homedir)/.cargo/registry
         displayName: Cache cargo cache folder
         continueOnError: true
-  - ${{ if parameters.cacheRegistry }}:
+  - ${{ if parameters.cacheTarget }}:
```
Comment (connorsmith256) [inline steps/rust-caching.yml:34]: Assuming this was a copy-pasta error

File: steps/rust-caching.yml [hunk 3]
```diff
@@ -25,17 +25,16 @@ steps:
           Write-Host "##vso[task.setvariable variable=homedir]$env:HOME"
         displayName: Get Home directory (windows)
         condition: eq(variables['Agent.OS'], 'Windows_NT')
-  - ${{ if parameters.cacheRegistry }}:
       - task: Cache@2
         inputs:
-          key: '"rust-${{ parameters.projectName }}-cache" | "$(Agent.OS)" | "${{ parameters.workingDirectory }}/Cargo.toml"'
+          key: '"rust-${{ parameters.projectName }}-cache" | "$(Agent.OS)" | ${{ parameters.workingDirectory }}/Cargo.toml'
```
Comment (connorsmith256) [inline steps/rust-caching.yml:30]: If I'm reading the docs correctly, double-quoted key segments are treated as fixed names. Leaving off the quotes should treat it as a path, so the contents of the file get hashed and used as part of the key

https://docs.microsoft.com/en-us/azure/devops/pipelines/release/caching?view=azure-devops#key-input

File: steps/rust-caching.yml (hunk 1 above)
Comment (thomastaylor312) [reply steps/rust-caching.yml:28]: This is needed for conditional inclusion of the block so it needs to stay

File: steps/rust-caching.yml (hunk 2 above)
Comment (thomastaylor312) [reply steps/rust-caching.yml:34]: Yeah. This actually could be part of the problem

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Just the one small change for the conditional block inclusion

**Review by thomastaylor312 — APPROVED** [review]

File: steps/rust-caching.yml (hunk 1 above)
Comment (thomastaylor312) [reply steps/rust-caching.yml:28]: Nvm I figured out why this isn’t needed

**Pull request merged**